### PR TITLE
Improve releasing during v0.11.0

### PR DESCRIPTION
**Pull request by jedevc**

### CODE REVIEW

**Comment by jedevc** [comment]
cc @matipan @gerhard this PR will fail until the CI runners can be updated to v0.11.0.

**Comment by gerhard** [comment]
I have just updated all workflows except those that need a dev-engine to use the new CI setup. In the new setup, the workflows request the Engine version, and this gets provisioned just-in-time, meaning no more coordination is needed with ops/infra. Let's see how well the new CI setup works in practice. cc @matipan 

---

The reason why we switched to the new CI setup instead of migrated slowly [is captured in this private discussion](https://linear.app/dagger/issue/INFRA-169/investigate-increase-in-aws-ci-account-cost-for-march-2024#comment-346c2b89).

**Comment by gerhard** [comment]
I am going to mark this ready for review so that the jobs trigger.

While waiting, will resolve the conflicts.

**Comment by gerhard** [comment]
Stopping here for now, next steps:
- Reduce `Engine & CLI / testdev` log output & find why it failed (too much output, couldn't see the reason)
- Fix `Engine & CLI . scan-engine` (needs v0.11.0, but we cannot bump it since all workflows using v0.10 will fail)
- `Engine & CLI / test` seems to be stuck after 25mins, not sure what the action item is jus yet

File: .github/workflows/_hack_make.yml [hunk 1]
```diff
@@ -56,37 +56,58 @@ jobs:
...
       - uses: actions/checkout@v4
       - uses: actions/setup-go@v5
         with:
           go-version: "1.21"
           cache-dependency-path: "ci/go.sum"
+      - name: Waiting for Dagger Engine to be ready...
+        run: |
+          ./hack/make engine:connect
+        env:
+          _EXPERIMENTAL_DAGGER_CLOUD_TOKEN: "p.eyJ1IjogIjFiZjEwMmRjLWYyZmQtNDVhNi1iNzM1LTgxNzI1NGFkZDU2ZiIsICJpZCI6ICIwYzhmMGY2Yy00YjY1LTRhODktYTI0YS0yN2NjNWNhNzNmNTcifQ.Em92UDP-KmcNd80Y-euTS2IrRkIE8qTisP3SXYkQi3c"
```
[the remark this answers is not in the document]
Comment (jedevc) [reply .github/workflows/_hack_make.yml:76]: Yeah, on reflection this is a pretty big issue. With v0.11.0, there's actually no way to have *only* traces sent to cloud, there's only one token for everything. So the options are:

- Upgrade to v0.11.0, no cloud traces for us
- Skip upgrading to v0.11.0, and upgrade when we fix this

This feels like something we should resolve in v0.11.1 - will spin out an issue.

**Review by vito — APPROVED** [review]

**Pull request merged**